File: Practice/28/C++/example28/example28/example28.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

void print_factorization(unsigned int n) {
    vector<int> nums;
    unsigned int a = n;
    bool go = true;


    int i = 1;
    if (a != 1)
    {
        i++;
    }
    while (go)
    {
        if (a % i == 0)
        {
            nums.push_back(i);
            a = a / i;
        }
        else {
            i++;
        }
        if (a == 1)
        {
            break;
        }
        //Чтобы выйти из цикла  go = false
       /* i++;*/
    }
    int count = 1;/// 3 3 3 37
    unsigned int templast = -1000;
    int zvezd = 0;
    for (int i = 0; i < nums.size()-1; i++)
    {

              //group

        if (nums[i] == nums[i + 1])
        {
            if (i == 0)
            {
                zvezd = 1;
            }
            count++;
        }
        if (count != 1 && nums[i] != nums[i + 1])
        {
            if (zvezd) {


                cout << nums[i] << "^" << count;
                zvezd = 0;
            }
            else {
                cout << "*" << nums[i] << "^" << count;
            }

            count = 1;
        }
        //solo
        if (templast != nums[i] && nums[i] != nums[i+1] && i != 0) // solo middle
        {
            cout << "*" << nums[i];
        }
        
        if (nums[i] != nums[i+1] && i == 0) //solo start
        {
            cout << nums[i];
        }

                
        if (nums[i] != nums[i+1] && i == nums.size() - 2) // solo finish
        {
            cout << "*" << nums[i+1];
        }
  



        templast = nums[i];
    }
}
int main(){

    unsigned int n;
    cin >> n;
    print_factorization(n);
}

File: Practice/28/Python/example28.py
def print_factorization(n:int)->None:
    nums = []
    a=n
    go = True
    i = 1
    if a != 1:
        i +=1
    while go:
        if a % i == 0:
            nums.append(i)
            a = a/i
        else:
            i +=1
        if a==1:
            break
    count = 1
    templast = -1000
    zvezd = 0
    j = 0
    while j<len(nums)-1:
        if nums[j]==nums[j+1]:
            if j == 0:
                zvezd = 1
            count +=1
        if count != 1 and nums[j] != nums[j+1] or j == len(nums)-2 and nums[j] == nums[j+1]:
            if zvezd:
                print(nums[j],"^",count,sep='',end='')
                zvezd = 0
            else:
                print("*",nums[j],"^",count,sep='',end='')
            count = 1
        if templast != nums[j] and nums[j] !=nums[j+1] and j != 0:
            print("*",nums[j],sep='',end='')
        if nums[j] != nums[j+1] and j == 0:
            print(nums[j],sep='',end='')
        if nums[j] != nums[j+1] and j == len(nums)-2:
            print("*",nums[j+1],sep='',end='')
        templast = nums[j]
        j +=1
n = int(input())
print_factorization(n)